### PR TITLE
Preference Based Ninjas + Fixes Ninja Jumpsuit (needs digi version still)

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Wouldn't it be neat if you could be your character as a midround ninja? I thought so, so I did this while trying to fix the ninja jumpsuit. 

When you roll Ninja, you will start as a bald-ass human, and be prompted if you want to be a your CHARACTER or a random fugly human with a random name (old way). If you hit YES, you will then become your CHARACTER. You will then be ASKED your CODENAME. By default your codename is RANDOMLY GENERATED as (TITLE SPECIES NAME). You can put whatever you want here. If you fail to input a name, you will become "Master Kobold" or something! 

https://user-images.githubusercontent.com/77420409/173706216-791c675f-4b8b-4201-8b95-145b2c274911.mp4

dreamseeker_xZeuSNIexJ

## How This Contributes To The Skyrat Roleplay Experience

It could be cool to have your alter-ego of your character moonlight as a space ninja!

## Changelog

:cl:
fix: Fixes the path for the ninja jumpsuit. Still no digi version, has a TG bug with digi selected.. 
add: You can now become your character as a SPACE NINJA.
/:cl:

### CODE REVIEW

File: code/modules/events/space_ninja.dm [hunk 1]
```diff
@@ -35,6 +35,21 @@
 	ninja.key = key
 	ninja.mind.add_antag_datum(/datum/antagonist/ninja)
 	spawned_mobs += ninja
+// SKYRAT EDIT ADDITION BEGIN: Preference Ninjas
+	var/loadme = tgui_input_list(ninja, "Do you wish to load your character slot?", "Load Character?", list("Yes!", "No, I want to be random!"), default = "No, I want to be random!", timeout = 60 SECONDS)
+	var/codename
+	switch(loadme)
```
Comment (Zonespace27) [inline code/modules/events/space_ninja.dm]: what is the point of this switch if it's literally just an if/else

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/events/space_ninja.dm (hunk 1 above)
Comment (nevimer) [reply code/modules/events/space_ninja.dm]: in my head this saves a whole microsecond of compute time but idk lol

File: code/modules/ninja/ninja_clothing.dm [hunk 2]
```diff
@@ -23,3 +23,9 @@
 	strip_delay = 12 SECONDS
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 	can_adjust = FALSE
+
+// SKYRAT EDIT ADDITION - Use TG sprites for missing assets
+/obj/item/clothing/under/syndicate/ninja/Initialize(mapload)
+	. = ..()
+	icon = 'icons/obj/clothing/under/syndicate.dmi'
```
Comment (Gandalf2k15) [inline code/modules/ninja/ninja_clothing.dm]: why aren't you just overriding this in modular?

Comment (nevimer) [reply code/modules/ninja/ninja_clothing.dm]: sure I can do that

Comment (nevimer) [reply code/modules/ninja/ninja_clothing.dm]: doneso

**Comment by YakumoChen** [comment]
I don't feel this fits the lore of space ninjas, heralding from the super secret spider clan, giving up their secrets to any old dumbshit from Nanotrasen crew to go fuck up the station...

**Comment by nevimer** [comment]
> I don't feel this fits the lore of space ninjas, heralding from the super secret spider clan, giving up their secrets to any old dumbshit from Nanotrasen crew to go fuck up the station...

The idea is you treat it like you would any other antag round. Non-canon, not the same as a station character, etc. Like when you roll assops, or tot, or changeling....

**Comment by YakumoChen** [comment]
My traitor rounds are canon, I'm not sure what you mean. Traitor and Ling (it's not you, it's a creature that is a sick copy of you) at least make sense in context. Not Assops (but I'm not sure how spawning for that works), but ninja doesn't make any sense as a station character, it feels like a breach of immersion, like playing station characters on Interdyne.

**Comment by nevimer** [comment]
> My traitor rounds are canon, I'm not sure what you mean. Traitor and Ling (it's not you, it's a creature that is a sick copy of you) at least make sense in context. Not Assops (but I'm not sure how spawning for that works), but ninja doesn't make any sense as a station character, it feels like a breach of immersion, like playing station characters on Interdyne.

Would you like a prompt to bring up the character slot and a READY button? That way you can load up your designated ninja character or what not without pressure?

**Comment by YakumoChen** [comment]
I'd prefer if this came with policy not to play with your station character (or at least rename your character ala Interdyne policy) if you're going to give the choice to be able to choose slots, but yes.

With spawners you can change your character slot in before you press "Yes" on "use loaded slot for character" so a picker isn't needed.

**Comment by nevimer** [comment]
> 



> I'd prefer if this came with policy not to play with your station character (or at least rename your character ala Interdyne policy) if you're going to give the choice to be able to choose slots, but yes.
> 
> With spawners you can change your character slot in before you press "Yes" on "use loaded slot for character" so a picker isn't needed.

Oh, this already forces you to rename your character. If you ignore the prompt to do so, it gives you a random-ish name.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**